securityscanutils/trivy_templates.go

```diff
@@ -0,0 +1,137 @@
+              "defaultConfiguration": {
+                "level": "{{ toSarifErrorLevel .Vulnerability.Severity }}"
+              }
+              {{- with $help_uri := .PrimaryURL -}}
+              ,
+              {{ $help_uri | printf "\"helpUri\": %q," -}}
+              {{- else -}}
+              ,
+              {{- end }}
+              "help": {
+                "text": {{ printf "Vulnerability %v\nSeverity: %v\nPackage: %v\nInstalled Version: %v\nFixed Version: %v\nLink: [%v](%v)" .VulnerabilityID .Vulnerability.Severity .PkgName .InstalledVersion .FixedVersion .VulnerabilityID .PrimaryURL | printf "%q"}},
+                "markdown": {{ printf "**Vulnerability %v**\n| Severity | Package | Installed Version | Fixed Version | Link |\n| --- | --- | --- | --- | --- |\n|%v|%v|%v|%v|[%v](%v)|\n" .VulnerabilityID .Vulnerability.Severity .PkgName .InstalledVersion .FixedVersion .VulnerabilityID .PrimaryURL | printf "%q"}}
+              },
+              "properties": {
+                "tags": [
+                  "vulnerability",
+                  "{{ .Vulnerability.Severity }}",
+                  {{ .PkgName | printf "%q" }}
+                ],
+                "precision": "very-high"
+              }
+            }
+            {{- end -}}
+         {{- end -}}
+          ]
+        }
+      },
+      "results": [
+    {{- $t_first := true }}
+    {{- range $result := . }}
+        {{- $filePath := .Target }}
+        {{- range $index, $vulnerability := .Vulnerabilities -}}
+          {{- if $t_first -}}
+            {{- $t_first = false -}}
+          {{ else -}}
+            ,
+          {{- end }}
+        {
+          "ruleId": {{ printf "%s: %s-%s %s" $result.Target .PkgName .InstalledVersion .VulnerabilityID | toJson }},
+          "ruleIndex": {{ $index }},
+          "level": "{{ toSarifErrorLevel $vulnerability.Vulnerability.Severity }}",
+          "message": {
+            "text": {{ endWithPeriod (escapeString $vulnerability.Description) | printf "%q" }}
+          },
+          "locations": [{
+            "physicalLocation": {
+              "artifactLocation": {
+                "uri": "{{ toPathUri $filePath }}",
+                "uriBaseId": "ROOTPATH"
+              }
+            }
+          }]
+        }
+        {{- end -}}
+      {{- end -}}
+      ],
+      "columnKind": "utf16CodeUnits",
+      "originalUriBaseIds": {
+        "ROOTPATH": {
+          "uri": "/"
+        }
+      }
+    }
+  ]
+}`
+
+// Create tempoarary file that contains the trivy template
+// Trivy CLI only accepts files as input for a template, so this is a workaround
+func GetTemplateFile(trivyTemplate string) (string, error) {
+	f, err := ioutil.TempFile("", "")
+	if err != nil {
+		return "", eris.Wrap(err, "Unable to create temporary file to write template to")
+	}
+	_, err = f.Write([]byte(trivyTemplate))
+	if err != nil {
+		return "", eris.Wrapf(err, "Unable to write template to file %s", f.Name())
+	}
+	return f.Name(), nil
```

Comment:
are we intentionally leaving the file open for writing?

Reply:
All writing is done here, I should close this file in this method.